src/app/admin/modals/create-unit-modal/create-unit-modal.component.ts

```diff
@@ -0,0 +1,58 @@
+// Co-authored-by: Mitchell Burcheri <mburcheri@deakin.edu.au>
+// Co-authored-by: Ray Guo <rguo@deakin.edu.au>
+
+import { Component, Inject, OnInit } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { analyticsService, Unit, alertService } from 'src/app/ajs-upgraded-providers';
+import { DoubtfireConstants } from 'src/app/config/constants/doubtfire-constants';
+
+// Angular component decorator
+@Component({
+  selector: 'df-create-break-modal',
+  templateUrl: 'create-unit-modal.component.html',
+  //styleUrls: ['create-unit-modal.component.scss'],
+})
+
+// CreateUnitModalComponent for exporting to other programs
+export class CreateUnitModalComponent implements OnInit {
+  // Defining variables
+  units: any = this.data.units;
+  unit: any = { code: null, name: null };
+  constructor(
+    // data being injected such as: CreateUnitModalComponent(analyticsService,Unit,alertService,DoubtfireConstants,MAT_DIALOG_DATA)
```

Comment:
This comment can be removed as we all know thats inject used for.

Reply:
Thanks it has now been removed